android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java

```diff
@@ -106,14 +106,7 @@ public boolean getLocationServicesEnabled()
 			}
 		}
 
-		// don't count the passive provider
-		for(String name : providerNames) {
-			if (name.equals(LocationManager.NETWORK_PROVIDER) || name.equals(LocationManager.GPS_PROVIDER)) {
-				return true;
-			}
-		}
-
-		return false;
+		return providerNames.size()!=0 & getLastKnownLocation()!=null;
```

Comment:
providerNames.size() is always >=1 because there is passive provider. 

But having passive provider doesn't guarantee there is a location available, and when there is a lastKnownLocation available doesn't guarantee there's other service provider available. That's why I didn't use short circuit &&. Does it make sense?